# Predicitng College Basketball Teams Total Wins 

## Importing Libraries 

In [1]:
import pandas as pd  
import numpy as np  
import matplotlib.pyplot as plt  
import seaborn as sns 
from sklearn.model_selection import train_test_split 
from sklearn.linear_model import LinearRegression
from sklearn import metrics
from statsmodels.formula.api import ols
%matplotlib inline

## Loading in Dataset

In [9]:
file = 'file:///Users/chrisfiorentine/Downloads/college-basketball-dataset%20(1)/cbb.csv'

### Turn Dataset into DataFrame

In [113]:
df = pd.read_csv(file)
df

TEAM CONF   G   W  ADJOE  ADJDE  BARTHAG  EFG_O  EFG_D   TOR  \
0     North Carolina  ACC  40  33  123.3   94.9   0.9531   52.6   48.1  15.4   
1          Wisconsin  B10  40  36  129.1   93.6   0.9758   54.8   47.7  12.4   
2           Michigan  B10  40  33  114.4   90.4   0.9375   53.9   47.7  14.0   
3         Texas Tech  B12  38  31  115.2   85.2   0.9696   53.5   43.0  17.7   
4            Gonzaga  WCC  39  37  117.8   86.3   0.9728   56.6   41.1  16.2   
...              ...  ...  ..  ..    ...    ...      ...    ...    ...   ...   
1752       Texas A&M  SEC  35  22  111.2   94.7   0.8640   51.4   46.9  19.2   
1753             LSU  SEC  35  28  117.9   96.6   0.9081   51.2   49.9  17.9   
1754       Tennessee  SEC  36  31  122.8   95.2   0.9488   55.3   48.1  15.8   
1755         Gonzaga  WCC  35  27  117.4   94.5   0.9238   55.2   44.8  17.1   
1756         Gonzaga  WCC  37  32  117.2   94.9   0.9192   57.0   47.1  16.1   

      ...  FTRD  2P_O  2P_D  3P_O  3P_D  ADJ_T   WAB  POSTSEASON  SEED  YEAR  
0     ...  30.4  53.9  44.6  32.7  36.2   71.7   8.6         2ND   1.0  2016  
1     ...  22.4  54.8  44.7  36.5  37.5   59.3  11.3         2ND   1.0  2015  
2     ...  30.0  54.7  46.8  35.2  33.2   65.9   6.9         2ND   3.0  2018  
3     ...  36.6  52.8  41.9  36.5  29.7   67.5   7.0         2ND   3.0  2019  
4     ...  26.9  56.3  40.0  38.2  29.0   71.5   7.7         2ND   1.0  2017  
...   ...   ...   ...   ...   ...   ...    ...   ...         ...   ...   ...  
1752  ...  27.6  52.5  45.7  32.9  32.6   70.3   1.9         S16   7.0  2018  
1753  ...  33.1  52.9  49.4  31.9  33.7   71.2   7.3         S16   3.0  2019  
1754  ...  34.9  55.4  44.7  36.7  35.4   68.8   9.9         S16   2.0  2019  
1755  ...  28.1  54.3  44.4  37.8  30.3   68.2   2.1         S16  11.0  2016  
1756  ...  29.1  58.2  44.1  36.8  35.0   70.5   4.9         S16   4.0  2018  

[1757 rows x 24 columns]

## Exploring Dataset

### Showing Dscriptive Statistics 

In [12]:
df.describe()

G            W        ADJOE        ADJDE      BARTHAG  \
count  1757.000000  1757.000000  1757.000000  1757.000000  1757.000000   
mean     31.523051    16.513375   103.542402   103.542459     0.493398   
std       2.602819     6.545571     7.304975     6.472676     0.255291   
min      24.000000     0.000000    76.700000    84.000000     0.007700   
25%      30.000000    12.000000    98.600000    98.900000     0.283700   
50%      31.000000    16.000000   103.100000   103.800000     0.474000   
75%      33.000000    21.000000   108.100000   108.000000     0.710600   
max      40.000000    38.000000   129.100000   124.000000     0.984200   

             EFG_O        EFG_D          TOR         TORD          ORB  ...  \
count  1757.000000  1757.000000  1757.000000  1757.000000  1757.000000  ...   
mean     50.120489    50.312806    18.591804    18.521286    29.277120  ...   
std       3.130430     2.859604     1.991637     2.108968     4.101782  ...   
min      39.400000    39.600000    12.400000    10.200000    15.000000  ...   
25%      48.100000    48.400000    17.200000    17.100000    26.600000  ...   
50%      50.000000    50.300000    18.500000    18.500000    29.400000  ...   
75%      52.100000    52.300000    19.800000    19.900000    31.900000  ...   
max      59.800000    59.500000    26.100000    28.000000    42.100000  ...   

               FTR         FTRD         2P_O         2P_D         3P_O  \
count  1757.000000  1757.000000  1757.000000  1757.000000  1757.000000   
mean     35.097894    35.373307    49.135970    49.298065    34.563517   
std       4.884599     5.900935     3.422136     3.288265     2.742323   
min      21.600000    21.800000    37.700000    37.700000    25.200000   
25%      31.700000    31.200000    46.900000    47.100000    32.600000   
50%      34.900000    34.900000    49.000000    49.300000    34.600000   
75%      38.300000    39.200000    51.400000    51.500000    36.400000   
max      51.000000    58.500000    62.600000    61.200000    44.100000   

              3P_D        ADJ_T          WAB        SEED         YEAR  
count  1757.000000  1757.000000  1757.000000  340.000000  1757.000000  
mean     34.744792    68.422254    -7.837109    8.791176  2017.002277  
std       2.369727     3.258920     6.988694    4.674090     1.415419  
min      27.100000    57.200000   -25.200000    1.000000  2015.000000  
25%      33.100000    66.400000   -13.000000    5.000000  2016.000000  
50%      34.700000    68.500000    -8.400000    9.000000  2017.000000  
75%      36.300000    70.400000    -3.100000   13.000000  2018.000000  
max      43.100000    83.400000    13.100000   16.000000  2019.000000  

[8 rows x 21 columns]

### Showing Bow Big Dataset is

In [14]:
df.shape #1757 rows 24 columns 

(1757, 24)

## Key for columns 

G = Games   
W = Wins   
ADJOE = Adjusted Offensive Efficency   
ADJDE = Adjusted Defensive Efficency  
BARTHAG = Power Rating (Chance of beating an average Division I team)  
EFG_O = Effective Field Goal Percentage Shot  
EFG_D = Effective Field Goal Percentage Allowed  
TOR = Turnover Percentage Allowed (Turnover Rate)  
TORD = Turnover Percentage Committed (Steal Rate)  
ORB = Offensive Rebound Percentage  
DRB = Defensive Rebound Percentage  
FTR = Free Throw Rate (How often the given team shoots Free Throws)  
FTRD = Free Throw Rate Allowed  
2P_O = Two-Point Shooting Percentage  
2P_D = Two-Point Shooting Percentage Allowed  
3P_O = Three-Point Shooting Percentage  
3P_D = Three-Point Shooting Percentage Allowed  
ADJ_T = Adjusted Tempo (An estimate of the tempo (possessions per 40 minutes) a team would have against the team that wants to play at an)  
WAB = Wins Above Bubble (The bubble refers to the cut off between making the NCAA March Madness Tournament and not making it)  
POSTSEASON = Round where the given team was eliminated or where their season ended  
SEED = Seed in the NCAA March Madness Tournament  
YEAR = Season

## Exploring Descriptive Statistics for Dependent and Independent Variables for All Years 

Very important to see how the data changes from year to year. If the data changes significantly, it may be a result of how the game has changed from year to year. This may be due to a rule changes, coaching styles, or other factors. If descriptive statistics show that the game has changed significantly, it may be the result of a time series and data may be invalid. 

### Loading Datasets for every year 

In [16]:
csv_2015 = pd.read_csv('file:///Users/chrisfiorentine/Downloads/college-basketball-dataset%20(1)/cbb15.csv')
csv_2016 = pd.read_csv('file:///Users/chrisfiorentine/Downloads/college-basketball-dataset%20(1)/cbb16.csv')
csv_2017 = pd.read_csv('file:///Users/chrisfiorentine/Downloads/college-basketball-dataset%20(1)/cbb17.csv')
csv_2018 = pd.read_csv('file:///Users/chrisfiorentine/Downloads/college-basketball-dataset%20(1)/cbb18.csv')
csv_2019 = pd.read_csv('file:///Users/chrisfiorentine/Downloads/college-basketball-dataset%20(1)/cbb19.csv')

### Turning Datasets into Data Frames

In [17]:
df_2015 = pd.DataFrame(csv_2015)
df_2016 = pd.DataFrame(csv_2016)
df_2017 = pd.DataFrame(csv_2017)
df_2018 = pd.DataFrame(csv_2018)
df_2019 = pd.DataFrame(csv_2019)

### Wins Stats Over all Years

In [18]:
wins_stats_2015 = df_2015['W'].describe() 
df_wins_stats_2015 = pd.DataFrame(wins_stats_2015)

In [19]:
wins_stats_2016 = df_2016['W'].describe()
df_wins_stats_2016 = pd.DataFrame(wins_stats_2016)

In [20]:
wins_stats_2017 = df_2017['W'].describe()
df_wins_stats_2017 = pd.DataFrame(wins_stats_2017)

In [21]:
wins_stats_2018 = df_2018['W'].describe()
df_wins_stats_2018 = pd.DataFrame(wins_stats_2018)

In [22]:
wins_stats_2019 = df_2019['W'].describe()
df_wins_stats_2019 = pd.DataFrame(wins_stats_2019)

In [23]:
wins_stats = df['W'].describe()
df_wins_stats = pd.DataFrame(wins_stats) #Stats for wins over all years

In [97]:
df_all_wins_stats = pd.concat([df_wins_stats_2015,df_wins_stats_2016, df_wins_stats_2017,df_wins_stats_2018,df_wins_stats_2019,df_wins_stats], axis = 1)
df_all_wins_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years

W           W           W           W           W            W
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean    15.666667   15.712251   17.002849   17.068376   17.113314    16.513375
std      6.698401    6.686006    6.456226    6.368305    6.374993     6.545571
min      0.000000    1.000000    2.000000    3.000000    3.000000     0.000000
25%     11.000000   10.000000   12.000000   12.000000   12.000000    12.000000
50%     15.000000   16.000000   17.000000   17.000000   17.000000    16.000000
75%     20.000000   21.000000   21.000000   21.000000   21.000000    21.000000
max     38.000000   35.000000   37.000000   36.000000   35.000000    38.000000

Average wins is fairly consistent over all years giving an average of about 16.5 wins per team 

### Games Played Stats

Important to show all seasons play the same amount of games 

In [27]:
games_stats = df['G'].describe()
df_game_stats = pd.DataFrame(games_stats) #Stats for all years

In [37]:
games_stats_2015 = df_2015['G'].describe()
df_game_stats_2015 = pd.DataFrame(games_stats_2015)

In [36]:
games_stats_2016 = df_2016['G'].describe()
df_game_stats_2016 = pd.DataFrame(games_stats_2016)

In [35]:
games_stats_2017 = df_2017['G'].describe()
df_game_stats_2017 = pd.DataFrame(games_stats_2017)

In [34]:
games_stats_2018 = df_2018['G'].describe()
df_game_stats_2018 = pd.DataFrame(games_stats_2018)

In [33]:
games_stats_2019 = df_2019['G'].describe()
df_game_stats_2019 = pd.DataFrame(games_stats_2019)

In [96]:
df_all_game_stats = pd.concat([df_game_stats_2015,df_game_stats_2016, df_game_stats_2017,df_game_stats_2018,df_game_stats_2019,df_game_stats], axis = 1)
df_all_game_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years

G           G           G           G           G            G
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean    31.333333   31.430199   31.538462   31.566952   31.745042    31.523051
std      2.660827    2.606607    2.611639    2.615436    2.514195     2.602819
min     24.000000   24.000000   25.000000   24.000000   26.000000    24.000000
25%     29.000000   30.000000   30.000000   30.000000   30.000000    30.000000
50%     31.000000   31.000000   31.000000   31.000000   31.000000    31.000000
75%     33.000000   33.000000   33.000000   33.000000   33.000000    33.000000
max     40.000000   40.000000   39.000000   40.000000   39.000000    40.000000

All seasons had about an average of about 31 games played each year with a total average of 31.5

### ADJOE Stats 

Adjusted Offensive Efficency 

In [43]:
adjoe_stats = df['ADJOE'].describe()
df_adjoe_stats = pd.DataFrame(adjoe_stats) #Stats for all years

In [44]:
adjoe_stats_2015 = df_2015['ADJOE'].describe()
df_adjoe_stats_2015 = pd.DataFrame(adjoe_stats_2015)

In [45]:
adjoe_stats_2016 = df_2016['ADJOE'].describe()
df_adjoe_stats_2016 = pd.DataFrame(adjoe_stats_2016)

In [46]:
adjoe_stats_2017 = df_2017['ADJOE'].describe()
df_adjoe_stats_2017 = pd.DataFrame(adjoe_stats_2017)

In [47]:
adjoe_stats_2018 = df_2018['ADJOE'].describe()
df_adjoe_stats_2018 = pd.DataFrame(adjoe_stats_2018)

In [48]:
adjoe_stats_2019 = df_2019['ADJOE'].describe()
df_adjoe_stats_2019 = pd.DataFrame(adjoe_stats_2019)

In [95]:
df_all_adjoe_stats = pd.concat([df_adjoe_stats_2015,df_adjoe_stats_2016, df_adjoe_stats_2017,df_adjoe_stats_2018,df_adjoe_stats_2019,df_adjoe_stats], axis = 1)
df_all_adjoe_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years 

ADJOE       ADJOE       ADJOE       ADJOE       ADJOE        ADJOE
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean   102.315385  103.866382  103.840456  104.354986  103.335977   103.542402
std      7.401505    7.462340    7.334709    7.175631    7.020019     7.304975
min     76.700000   82.900000   84.000000   86.200000   83.700000    76.700000
25%     97.900000   99.050000   98.850000   99.050000   98.600000    98.600000
50%    102.100000  103.200000  103.100000  104.100000  103.100000   103.100000
75%    106.500000  109.000000  108.300000  108.800000  107.700000   108.100000
max    129.100000  123.300000  126.800000  128.400000  123.400000   129.100000

ADJOE is consistent over all years with a total average ADJOE of about 103.5

### ADJDE Stats 

Adjusted Defensive Efficency 

In [53]:
df_adjde_stats = pd.DataFrame(df['ADJDE'].describe()) #Stats for all years

In [54]:
adjde_stats_2015 = df_2015['ADJDE'].describe()
df_adjde_stats_2015 = pd.DataFrame(adjde_stats_2015)

In [55]:
adjde_stats_2016 = df_2016['ADJDE'].describe()
df_adjde_stats_2016 = pd.DataFrame(adjde_stats_2016)

In [56]:
adjde_stats_2017 = df_2017['ADJDE'].describe()
df_adjde_stats_2017 = pd.DataFrame(adjde_stats_2017)

In [57]:
adjde_stats_2018 = df_2018['ADJDE'].describe()
df_adjde_stats_2018 = pd.DataFrame(adjde_stats_2018)

In [58]:
adjde_stats_2019 = df_2019['ADJDE'].describe()
df_adjde_stats_2019 = pd.DataFrame(adjde_stats_2019)

In [94]:
df_all_adjde_stats = pd.concat([df_adjde_stats_2015,df_adjde_stats_2016, df_adjde_stats_2017,df_adjde_stats_2018,df_adjde_stats_2019,df_adjde_stats], axis = 1)
df_all_adjde_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years

ADJDE       ADJDE       ADJDE       ADJDE       ADJDE        ADJDE
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean   102.311681  103.866382  103.841880  104.357265  103.336261   103.542459
std      6.537436    6.325834    6.591266    6.300789    6.452597     6.472676
min     84.000000   87.800000   86.300000   86.700000   85.200000    84.000000
25%     97.750000   99.100000   99.500000   99.850000   98.700000    98.900000
50%    102.600000  103.800000  103.700000  104.400000  104.000000   103.800000
75%    107.100000  108.400000  108.250000  108.300000  107.800000   108.000000
max    120.000000  118.200000  124.000000  123.000000  119.200000   124.000000

ADJDE is consistent over all years with a total average ADJDE of about 103.5

### Tempo Stats

Important to see if the pace of the games played stays consistent for all years as it may be a result of rule change, different coaching styles or other factors resulting in a time series affecting the data and making the data invalid.

In [62]:
df_tempo_stats = pd.DataFrame(df['ADJ_T'].describe()) #Stats for all years

In [67]:
df_tempo_stats_2015 = pd.DataFrame(df_2015['ADJ_T'].describe()) # Same code as before just written in one line

In [68]:
df_tempo_stats_2016 = pd.DataFrame(df_2016['ADJ_T'].describe())

In [69]:
df_tempo_stats_2017 = pd.DataFrame(df_2017['ADJ_T'].describe())

In [70]:
df_tempo_stats_2018 = pd.DataFrame(df_2018['ADJ_T'].describe())

In [71]:
df_tempo_stats_2019 = pd.DataFrame(df_2019['ADJ_T'].describe())

In [93]:
df_all_tempo_stats = pd.concat([df_tempo_stats_2015,df_tempo_stats_2016, df_tempo_stats_2017,df_tempo_stats_2018,df_tempo_stats_2019,df_tempo_stats], axis = 1)
df_all_tempo_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years

ADJ_T       ADJ_T       ADJ_T       ADJ_T       ADJ_T        ADJ_T
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean    64.936467   69.100855   69.481766   69.419373   69.168555    68.422254
std      2.686141    2.780478    2.917897    2.681646    2.694397     3.258920
min     57.200000   61.900000   60.300000   60.500000   60.700000    57.200000
25%     63.300000   67.200000   67.700000   67.500000   67.300000    66.400000
50%     64.900000   68.900000   69.600000   69.400000   69.000000    68.500000
75%     66.600000   70.850000   71.100000   71.000000   70.800000    70.400000
max     77.300000   81.900000   83.300000   83.400000   79.100000    83.400000

2015 has an average tempo of 64 possensions per game where every other year has average of 69.   
Shown later in hypothesis test to see if 2015 tempo is statistcally different enough to rule out. 

#### NOTE 

A rule-change after the 2015 had a direct affect on the tempo of the game. The rule-change changed the shot clock from 35 to 30 seconds. This means that each time the possesion changes, the team with the ball has a maximun of 30 seconds to shoot the ball instead of 35. This sped up the game and changed the tempo. Before the rule-change, teams averaged about 65 possesions per 40 minutes. After the rule-change, speeding up the shotclock, the average posseions per game spiked to 69 per 40 minutes as seen in the data.   

### Turnover Stats

Turnover Rate

In [76]:
df_tor_stats = pd.DataFrame(df['TOR'].describe()) #Stats for all years

In [77]:
df_tor_stats_2015 = pd.DataFrame(df_2015['TOR'].describe())

In [78]:
df_tor_stats_2016 = pd.DataFrame(df_2016['TOR'].describe())

In [79]:
df_tor_stats_2017 = pd.DataFrame(df_2017['TOR'].describe())

In [80]:
df_tor_stats_2018 = pd.DataFrame(df_2018['TOR'].describe())

In [81]:
df_tor_stats_2019 = pd.DataFrame(df_2019['TOR'].describe())

In [82]:
df_all_tor_stats = pd.concat([df_tor_stats_2015,df_tor_stats_2016, df_tor_stats_2017,df_tor_stats_2018,df_tor_stats_2019,df_tor_stats], axis = 1)
df_all_tor_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years

TOR         TOR         TOR         TOR         TOR          TOR
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean    19.125641   18.184046   18.594017   18.448718   18.606516    18.591804
std      2.070562    1.898790    1.893913    1.911168    2.066711     1.991637
min     12.400000   13.600000   13.900000   13.500000   13.500000    12.400000
25%     17.800000   16.800000   17.400000   17.200000   17.200000    17.200000
50%     19.200000   18.100000   18.500000   18.500000   18.500000    18.500000
75%     20.300000   19.450000   19.800000   19.600000   19.800000    19.800000
max     26.100000   25.400000   25.900000   24.700000   25.100000    26.100000

Turnover Rate is consistent over all years with a total average Turnover Rate of about 18.6

### Free Throw Statistics

Free Throw Rate

In [83]:
df_ft_stats = pd.DataFrame(df['FTR'].describe()) #Stats for all years

In [86]:
df_ft_stats_2015 = pd.DataFrame(df_2015['FTR'].describe())

In [87]:
df_ft_stats_2016 = pd.DataFrame(df_2016['FTR'].describe())

In [88]:
df_ft_stats_2017 = pd.DataFrame(df_2017['FTR'].describe())

In [89]:
df_ft_stats_2018 = pd.DataFrame(df_2018['FTR'].describe())

In [90]:
df_ft_stats_2019 = pd.DataFrame(df_2019['FTR'].describe())

In [92]:
df_all_ft_stats = pd.concat([df_ft_stats_2015,df_ft_stats_2016, df_ft_stats_2017,df_ft_stats_2018,df_ft_stats_2019,df_ft_stats], axis = 1)
df_all_ft_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years

FTR         FTR         FTR         FTR         FTR          FTR
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean    37.043305   36.638177   35.328205   33.537607   32.954391    35.097894
std      4.747210    4.610320    4.493351    4.485782    4.708719     4.884599
min     26.100000   25.500000   24.000000   21.600000   21.900000    21.600000
25%     33.550000   33.300000   32.450000   30.400000   29.900000    31.700000
50%     36.600000   36.800000   35.100000   33.300000   33.300000    34.900000
75%     40.300000   40.200000   38.250000   36.450000   35.900000    38.300000
max     51.000000   49.400000   46.000000   45.800000   48.100000    51.000000

Free Throw Rate steadily drops each year from 37 to 32.95 with an average Free Throw Rate of 35 

### Offensive Rebound Statistics 

Offensive Rebound Rate

In [132]:
df_ORB_stats = pd.DataFrame(df['ORB'].describe()) #Stats for all years 

In [133]:
df_ORB_stats_2015 = pd.DataFrame(df_2015['ORB'].describe())

In [134]:
df_ORB_stats_2016 = pd.DataFrame(df_2016['ORB'].describe())

In [135]:
df_ORB_stats_2017 = pd.DataFrame(df_2017['ORB'].describe())

In [136]:
df_ORB_stats_2018 = pd.DataFrame(df_2018['ORB'].describe())

In [137]:
df_ORB_stats_2019 = pd.DataFrame(df_2019['ORB'].describe())

In [138]:
df_all_ORB_stats = pd.concat([df_ORB_stats_2015,df_ORB_stats_2016, df_ORB_stats_2017,df_ORB_stats_2018,df_ORB_stats_2019,df_ORB_stats], axis = 1)
df_all_ORB_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years

ORB         ORB         ORB         ORB         ORB          ORB
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean    30.862963   29.618234   29.120798   28.543590   28.245892    29.277120
std      3.943029    4.109475    4.088345    3.922602    3.938373     4.101782
min     19.300000   17.700000   15.000000   18.000000   15.900000    15.000000
25%     28.650000   26.950000   26.750000   25.900000   25.400000    26.600000
50%     30.800000   29.600000   29.300000   28.700000   28.300000    29.400000
75%     33.700000   32.200000   31.600000   31.300000   31.000000    31.900000
max     42.100000   42.000000   41.300000   38.800000   38.700000    42.100000

Offensive Rebound Rate is consistent over all years with average of 29.3

### Defensive Rebound Stats

Defensive Rebound Rate

In [126]:
df_DRB_stats = pd.DataFrame(df['DRB'].describe()) #Stats for all years 

In [127]:
df_DRB_stats_2015 = pd.DataFrame(df_2015['DRB'].describe())

In [128]:
df_DRB_stats_2016 = pd.DataFrame(df_2016['DRB'].describe())

In [129]:
df_DRB_stats_2017 = pd.DataFrame(df_2017['DRB'].describe())

In [130]:
df_DRB_stats_2018 = pd.DataFrame(df_2018['DRB'].describe())

In [131]:
df_DRB_stats_2019 = pd.DataFrame(df_2019['ORB'].describe())

In [139]:
df_all_DRB_stats = pd.concat([df_DRB_stats_2015,df_DRB_stats_2016, df_DRB_stats_2017,df_DRB_stats_2018,df_DRB_stats_2019,df_DRB_stats], axis = 1)
df_all_DRB_stats # Combining all data frames on one data frame in order from 2015 to 2019 then all years

DRB         DRB         DRB         DRB         ORB          DRB
count  351.000000  351.000000  351.000000  351.000000  353.000000  1757.000000
mean    31.090028   29.821083   29.298575   28.713105   28.245892    29.467445
std      2.912875    2.873775    2.869380    2.998533    3.938373     3.061786
min     22.400000   21.700000   21.700000   18.400000   15.900000    18.400000
25%     29.200000   27.900000   27.300000   26.850000   25.400000    27.300000
50%     31.200000   29.800000   29.300000   28.700000   28.300000    29.400000
75%     32.900000   31.500000   31.250000   30.700000   31.000000    31.500000
max     40.400000   38.800000   37.600000   37.800000   38.700000    40.400000

Defensive Rebound Rate is fairly consistent over all years with a total average of 29.5

#### Note found in descriptive statistics 

The count(number of teams) for 2019 is 353 and every other year is 351. 2 teams were added in 2019. However, this will not affect the model and they will be added into the model. This is okay to do because the stats taken from the games those teams played will still help the model to predict wins. 

## Trimming Down Data

### Avoiding Multicollinearity

Multicollinearity is important to avoid because we do not want to add unnecessary variables to our model. If we do not check for multicollinearity in our variables, it may lead to a large number of coefficents and may confuse our model. 

In [142]:
df.corr() # Displays how each column is correlated with each other in a Data Frame

G         W     ADJOE     ADJDE   BARTHAG     EFG_O     EFG_D  \
G        1.000000  0.708838  0.613432 -0.606166  0.688059  0.346425 -0.481942   
W        0.708838  1.000000  0.754532 -0.690753  0.814512  0.617839 -0.609144   
ADJOE    0.613432  0.754532  1.000000 -0.528576  0.870686  0.732683 -0.333693   
ADJDE   -0.606166 -0.690753 -0.528576  1.000000 -0.852432 -0.221381  0.792320   
BARTHAG  0.688059  0.814512  0.870686 -0.852432  1.000000  0.543153 -0.627696   
EFG_O    0.346425  0.617839  0.732683 -0.221381  0.543153  1.000000 -0.120335   
EFG_D   -0.481942 -0.609144 -0.333693  0.792320 -0.627696 -0.120335  1.000000   
TOR     -0.336407 -0.467073 -0.601024  0.219779 -0.472329 -0.367975  0.101070   
TORD     0.065020  0.138805 -0.116231 -0.234615  0.054377 -0.144287 -0.020831   
ORB      0.261046  0.296395  0.261351 -0.294066  0.310917 -0.147990 -0.341636   
DRB     -0.184134 -0.366715 -0.266665  0.347646 -0.337804 -0.319901  0.172261   
FTR      0.090549  0.126931  0.063637 -0.125265  0.112072 -0.083514 -0.205807   
FTRD    -0.279593 -0.329245 -0.382290  0.180823 -0.327932 -0.404445  0.107161   
2P_O     0.339290  0.585806  0.646011 -0.251354  0.512045  0.893530 -0.148439   
2P_D    -0.439340 -0.529558 -0.328822  0.728488 -0.588121 -0.104079  0.907933   
3P_O     0.225821  0.432743  0.579193 -0.102715  0.386597  0.763028 -0.043770   
3P_D    -0.349726 -0.485485 -0.198275  0.564135 -0.427750 -0.100545  0.722404   
ADJ_T   -0.040433 -0.016057  0.070476  0.227852 -0.079611  0.120142  0.273412   
WAB      0.666595  0.905029  0.851663 -0.809486  0.941776  0.562904 -0.629864   
SEED    -0.607807 -0.612284 -0.791692  0.706046 -0.832709 -0.275137  0.386928   
YEAR     0.052233  0.091829  0.048861  0.055367  0.000228  0.196195  0.211657   

              TOR      TORD       ORB  ...       FTR      FTRD      2P_O  \
G       -0.336407  0.065020  0.261046  ...  0.090549 -0.279593  0.339290   
W       -0.467073  0.138805  0.296395  ...  0.126931 -0.329245  0.585806   
ADJOE   -0.601024 -0.116231  0.261351  ...  0.063637 -0.382290  0.646011   
ADJDE    0.219779 -0.234615 -0.294066  ... -0.125265  0.180823 -0.251354   
BARTHAG -0.472329  0.054377  0.310917  ...  0.112072 -0.327932  0.512045   
EFG_O   -0.367975 -0.144287 -0.147990  ... -0.083514 -0.404445  0.893530   
EFG_D    0.101070 -0.020831 -0.341636  ... -0.205807  0.107161 -0.148439   
TOR      1.000000  0.103437  0.134433  ...  0.161369  0.316898 -0.288945   
TORD     0.103437  1.000000  0.118496  ...  0.111844  0.392412 -0.079076   
ORB      0.134433  0.118496  1.000000  ...  0.280598  0.136233 -0.091259   
DRB      0.188585  0.289078  0.018964  ...  0.082964  0.235665 -0.299565   
FTR      0.161369  0.111844  0.280598  ...  1.000000  0.203173 -0.040561   
FTRD     0.316898  0.392412  0.136233  ...  0.203173  1.000000 -0.390594   
2P_O    -0.288945 -0.079076 -0.091259  ... -0.040561 -0.390594  1.000000   
2P_D     0.091686  0.027203 -0.341291  ... -0.202888  0.088794 -0.119930   
3P_O    -0.333397 -0.177276 -0.143735  ... -0.093085 -0.267869  0.402768   
3P_D     0.066363 -0.106798 -0.197189  ... -0.108273  0.103091 -0.138731   
ADJ_T   -0.102687 -0.065216 -0.105308  ...  0.039090 -0.000793  0.163304   
WAB     -0.470286  0.094765  0.332275  ...  0.146079 -0.333798  0.530586   
SEED     0.291587  0.057411 -0.314659  ...  0.091152  0.250020 -0.235655   
YEAR    -0.054840 -0.051754 -0.217610  ... -0.326780 -0.273743  0.245197   

             2P_D      3P_O      3P_D     ADJ_T       WAB      SEED      YEAR  
G       -0.439340  0.225821 -0.349726 -0.040433  0.666595 -0.607807  0.052233  
W       -0.529558  0.432743 -0.485485 -0.016057  0.905029 -0.612284  0.091829  
ADJOE   -0.328822  0.579193 -0.198275  0.070476  0.851663 -0.791692  0.048861  
ADJDE    0.728488 -0.102715  0.564135  0.227852 -0.809486  0.706046  0.055367  
BARTHAG -0.588121  0.386597 -0.427750 -0.079611  0.941776 -0.832709  0.000228  
EFG_O   -0.104079  0.763028 -0.100545  0.120142  0.562904 -0.275137  0.196195  


### Notes found in correlation

ADJOE and EFG_O have a correlation of 0.73. This correlation is too close to use in the same model. Since ADJOE has a closer correlation to Wins, ADJOE will be used in the model.  
ADJDE and EFG_D have a correlation of 0.79. ADJDE and 2P_D have a correlation of 0.73. These correlations are too close to use in the same model. Since ADJDE has the closest correlation to Wins, ADJDE will be used in the model.  

### Dropping Multicollinear Columns 

In [ ]:
df2 = df.drop('EFG_O')